hax/helper/configure.py

```diff
@@ -135,9 +136,12 @@ def get_join_ip() -> str:
         ]),
                           env=self._get_pythonic_env())
 
+        machine_id = Conf.machine_id
```

Comment:
Can we use our home-made ConfStoreProvider from `hare_mp` package instead?

Reasons:
1. It will make the code testable
2. It will make this code more coherent to other pieces that communicate to ConfStore
3. It works (in case it doesn't, we know how to fix it quick).

Reply:
Done